### PR TITLE
Add failing test for nested properties with booleans

### PR DESCRIPTION
**Pull request by josemarluedke**
This adds a test for a bug I encountered today.

cc/ @snewcomer

### CODE REVIEW

File: tests/dummy/app/templates/components/changeset-form.hbs [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 <h2>Changeset Email: <span data-test-changeset-user-email>{{this.changeset.user.email}}</span></h2>
 <h2>Changeset cid: <span data-test-changeset-cid>{{this.changeset.cid}}</span></h2>
+<h2>Changeset notifications email: <span data-test-changeset-notifications-email>{{this.changeset.notifications.email}}</span></h2>
+<h2>Changeset notifications sms: <span data-test-changeset-notifications-sms>{{this.changeset.notifications.sms}}</span></h2>
 <br />
 <h2>Model Email: <span data-test-model-user-name>{{this.model.user.name}}</span></h2>
 <h2>Model Email: <span data-test-model-user-email>{{this.model.user.email}}</span></h2>
 <h2>Model cid: <span data-test-model-cid>{{this.model.cid}}</span></h2>
+<h2>Model notifications email: <span data-test-model-notifications-email>{{this.model.notifications.email}}</span></h2>
+<h2>Model notifications sms: <span data-test-model-notifications-sms>{{this.model.notifications.sms}}</span></h2>
```
Comment (snewcomer) [inline tests/dummy/app/templates/components/changeset-form.hbs]: Can we confirm that using `changeset-get` helper in `master` makes the tests pass?

Comment (josemarluedke) [reply tests/dummy/app/templates/components/changeset-form.hbs]: Confirmed. Rebased with master and changed to use `changeset-get`. Bug still there.

**Comment by snewcomer** [comment]
@josemarluedke I think I know what we can do.  We check if the old content value is equal to what we want to set and if so, don't do anything.  I'll see if we can refactor this.

**Comment by snewcomer** [comment]
#416 partially fixed your issues.  I think there might be something else we need to do!

**Comment by snewcomer** [comment]
Ok everything should be g2g on master!  Can't thank you enough for your contributions.  These were also issues in 2.x and 1.x so these are very nice additions!

**Comment by josemarluedke** [comment]
Rebased with master, tests did pass locally. Thank you @snewcomer for working on fixing this.

Should we get this tests merged anyway?

**Pull request merged**